arcaflow_plugin_rtla/rtla_schema.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+import typing
+from dataclasses import dataclass
+from arcaflow_plugin_sdk import plugin, schema
+
+
+def params_to_flags(params: dict) -> str:
+    result = []
+    for key, value in params.items():
+        if not value:
+            continue
+        if isinstance(value, bool):
+            result.append(f"-{key}")
+        elif isinstance(value, list):
+            result.append(f"-{key}")
+            result.append(str(",".join(str(i) for i in value)))
+        else:
+            result.append(f"-{key}")
+            result.append(str(value))
+    return result
+
+
+@dataclass
+class TimerlatInputParams:
+    period: typing.Annotated[
+        typing.Optional[int],
+        schema.name("timerlat period"),
+        schema.description("Timerlat period in μs"),
+    ] = None
+    cpus: typing.Annotated[
+        typing.Optional[typing.List[int]],
+        schema.name("cpus"),
+        schema.description("Run the tracer only on the given cpus"),
+    ] = None
+    house_keeping: typing.Annotated[
+        typing.Optional[typing.List[int]],
+        schema.id("house-keeping"),
+        schema.name("house-keeping cpus"),
+        schema.description("Run rtla control threads only on the given cpus"),
+    ] = None
+    duration: typing.Annotated[
+        typing.Optional[int],
+        schema.name("timerlat duration seconds"),
+        schema.description("Duration of the session in seconds"),
+    ] = None
+    nano: typing.Annotated[
+        typing.Optional[bool],
+        schema.name("display nanoseconds"),
+        schema.description("Display data in nanoseconds"),
+    ] = None
+    bucket_size: typing.Annotated[
+        typing.Optional[int],
+        schema.id("bucket-size"),
+        schema.name("histogram bucket size"),
+        schema.description("Set the histogram bucket size (default 1)"),
+    ] = None
+    entries: typing.Annotated[
+        typing.Optional[int],
+        schema.name("histogram entries"),
+        schema.description("Set the number of entries of the histogram (default 256)"),
+    ] = None
+    user_threads: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("user-threads"),
+        schema.name("use user threads"),
+        schema.description(
+            "Use rtla user-space threads instead of kernel-space timerlat threads"
+        ),
+    ] = None
+
+    def to_flags(self) -> str:
+        return params_to_flags(
+            {
+                "p": self.period,
+                "c": self.cpus,
+                "H": self.house_keeping,
+                "d": self.duration,
+                "n": self.nano,
+                "b": self.bucket_size,
+                "E": self.entries,
+                "u": self.user_threads,
+            }
+        )
```

Comment:
This is a little weird. In other plugins, the Arca schema has been matched to the actual long command option names -- is that not practical here?

Reply:
This is a bug apparently with the wrapped `rtla` tool. At least for one parameter (I think it was `cpus`) the long format of the parameter was getting rejected as bad syntax. I think this is already fixed in a later rtla build, but this is what we have in a centos stream 9 base.

Reply:
That's ... awkward. I wonder what their codebase is that `--cpus` works differently than `-c`? 😦